sycl/source/detail/pi.cpp

```diff
@@ -313,9 +311,13 @@ std::vector<std::pair<std::string, backend>> findPlugins() {
       }
       if (!EsimdCpuFound && (Backend == backend::ext_intel_esimd_emulator ||
                              Backend == backend::all)) {
-        PluginNames.emplace_back(__SYCL_ESIMD_EMULATOR_PLUGIN_NAME,
-                                 backend::ext_intel_esimd_emulator);
-        EsimdCpuFound = true;
+        std::string env = std::string(std::getenv("SYCL_DEVICE_FILTER"));
+        std::transform(env.begin(), env.end(), env.begin(), ::tolower);
+        if (env.find("esimd_emulator") != std::string::npos) {
+          PluginNames.emplace_back(__SYCL_ESIMD_EMULATOR_PLUGIN_NAME,
```

Comment:
The rationale behind this is that we would like to enable ESIMD_EMULATOR only if the backend itself appears in the filter. For example, when SYCL_DEVICE_FILTER=gpu then this would also include the esimd_emulator. Another example, if I set SYCL_DEVICE_FILTER=0 then this also includes esimd_emulator and it also causes some test failures. So, since ESIMD_EMULATOR is special in the sense that it does not run SYCL kernel, we though it would be nice to impose on the application that the entire string "ESIMD_EMULATOR"(case does not matter) appears explicitly in the filter and only then include it.